demisto_sdk/commands/common/docker_helper.py

```diff
@@ -224,6 +224,17 @@ def copy_files_container(
                 with open(tar_file_path.name, "rb") as byte_file:
                     container.put_archive("/", byte_file.read())
 
+    def remove_container(self, name, **kwargs):
+        container: docker.models.containers.Container = (
+            init_global_docker_client().containers.get(name)
+        )
+        container.remove()
+
+    @retry(
+        times=3,
+        exceptions=(requests.exceptions.ConnectionError, requests.exceptions.Timeout),
+        recover=remove_container,
```

Comment:
@GuyAfik 

> if we failed to create the container due to connection errors, why would we need to remove it at all?

To avoid errors like this, the container probably created and for some (Timeout) exception it fails but the container name already exist and causes conflict errors.

> Also adding a recover function is kind of an abuse to the retry decorator IMO.

So what do you suggested?

Reply:
i think you should recover inside the `create_container` method rather than the `retry` decorator.

If you have ConnectionError exception, no way a container will be created, as to timeout, it might be possible that docker can create the container but won't provide any response about it. 

why do you need to use the `retry` decorator if you recover in case of an exception?
Don't you need to recover here only when you get 409?